libs/math/include/math/one_hot.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/assert.hpp"
+
+#include <cassert>
+
+namespace fetch {
+namespace math {
+
+/**
+ * One hot function
+ * @tparam ArrayType
+ * @param ret
+ * @param indices
+ * @param depth
+ * @param on_value
+ * @param off_value
+ * @param axis
+ */
+template <typename ArrayType>
+void OneHot(ArrayType &ret, ArrayType const &indices, typename ArrayType::SizeType depth,
+            typename ArrayType::Type on_value  = typename ArrayType::Type{1.0},
+            typename ArrayType::Type off_value = typename ArrayType::Type{0.0}, SizeType axis = 0)
```

Comment:
I can imagine using this function and specifying axis much more often than specifying the on value and off value. Since C++ syntax requires all previous optional arguments to be filled out up until the last one you set, it makes more sense to have axis be the first optional argument, and then have on_value and off_value after that